platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/OperationType.java

```diff
@@ -166,5 +166,20 @@ public enum OperationType {
      * @since 8.4
      */
     @Incubating
-    PROBLEMS
+    PROBLEMS,
+
+    /**
+     * Flag for the topmost progress event.
+     * <p>
+     * Using this operation type is useful for capturing the build failure details from the finish event.
+     *
+     * The following events are currently issued for this operation type.
+     * <ul>
+     *     <li>{@link StartEvent}</li>
+     *     <li>{@link FinishEvent}</li>
+     * </ul>
+     * @since 8.12
+     */
+    @Incubating
+    ROOT
```

Comment:
I added a dedicated category for the root operation to allow (tooling api) clients can subscribe only for that. We know that relying on the `GENERIC` category is not enough as some IDEs cannot subscribe to it due to the inherent performance penalty. Besides, we already have a plan to have more fine-grained operation types.